fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -3272,4 +3272,89 @@ public IndexState needRebuildIndex(Index index, long recordCount, boolean indexO
             assertTrue(recordStore.getRecordStoreState().getStoreHeader().getOmitUnsplitRecordSuffix());
         }
     }
+
+    @Test
+    public void testIndexStateFormatVersionUpgrade() {
+        final RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestNoIndexesProto.getDescriptor());
+        try (FDBRecordContext context = openContext()) {
+            recordStore = FDBRecordStore.newBuilder()
+                    .setContext(context)
+                    .setMetaDataProvider(builder)
+                    .setKeySpacePath(path)
+                    .setFormatVersion(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1)
+                    .create();
+            assertEquals(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1, recordStore.getFormatVersion());
+            commit(context);
```

Comment:
I think this test still doesn't validate that if there are any indexes a non-`READABLE` state that the code doesn't upgrade it on old format versions (which was the behavior alluded to here). I don't think the code still has that bug, but it would be nice if the tests covered it.

Reply:
done - take a look, I hope it's what you meant.

Reply:
Ah, yes, perfect.